docs/SPEC.md

```diff
@@ -3264,6 +3264,55 @@ from(bucket: "telegraf/autogen"):
     |> range(start: -7d)
     |> exponentialMovingAverage(n: 5, columns: ["_value"])
 ```
+
+##### Holt Winters
+
+Holt Winters applies the Holt-Winters damped prediction method with Nelder-Mead optimization to the given dataset.
+In a table, the dataset is composed of the values in the `timeColumn` and in the value `column`.
+The Holt-Winters method predicts `n` seasonally adjusted values for the specified `column`.
+If the data presents no seasonality, the user can specify `s = 0`.
+The user can request to include the fitted data (on the given dataset) in the result by setting `withFit = true`.
+The `n` predicted values occur at the `interval` specified by the user.
+If your `interval` is `6m` and `n` is `3`, you will receive three predicted values that are each six minutes apart.
+`s` is the seasonal pattern parameter and delimits the length of a seasonal pattern according to `interval`.
+If your `interval` is `2m` and `s` is `3`, then the seasonal pattern occurs every six minutes, that is, every three data points.
+
+Before processing the data, Holt Winters will cleanse it according to these rules:
+ - rows that have a null timestamp get discarded;
+ - rows that have a null value are considered invalid, but used by the algorithm;
+
+HoltWinters supposes to work with evenly spaced values in time, so:
+ - `interval` is used to divide the data in time buckets;
+ - if many values are in the same bucket, the first one is selected, the others are skipped;
+ - if no value is present for a bucket, that is considered as an invalid value (and treated like null values).
+
+Parameters:
+
+| Name        | Type     | Description
+| ----        | ----     | -----------
+| n           | int      | N specifies the number of values to predict.
+| s           | int      | S specifies the number of points that are in a season. Defaults to `0`.
+| interval    | duration | Interval is the interval between two data points.
+| withFit     | bool     | WithFit specifies if the fitted data should be returned. Defaults to `false`.
+| timeColumn  | string   | TimeColumn specifies the time column for the dataset. Defaults to `"_time"`.
+| column      | string   | Column specifies the value column for the dataset. Defaults to `"_value"`.
+
+By default, HoltWinters uses the first value per time bucket.
+The user can use windowing and selectors/aggregates to specify a different behavior.
+
+Example:
+
+```
+from(bucket: "waterhouse/autogen")
+    |> range(start: -7y)
+    |> filter(fn: (r) => r._field == "water_level")
+    |> window(every: 379m)
+    |> first()
+    // We need to "un-window", otherwise the Holt-Winters algorithm would run
+    // for each window separately.
+    |> window(every: inf)
```

Comment:
I am not using `aggregateWindow` because `first` is not an aggregator.

That shouldn't work because it doesn't have the `column` parameter.

Anyways, I'll add an example with `mean`.

Reply:
I see, I missed that, being a selector, `first` has the `column` parameter 👍

Reply:
Still  remains that you cannot specify  the `offset` in `aggregateWindow`. So maybe, it is better to show a complete example as the one above. What do you think?